numba/tests/test_hashing.py

```diff
@@ -254,16 +254,18 @@ def test_bool(self):
 
     def test_ints(self):
         minmax = []
+        # Temporarily set promotions state to legacy,
+        # to ensure overflow logic works
+        initial_state = np._get_promotion_state()
+        np._set_promotion_state("legacy")
```

Comment:
As far as this particular PR is concerned, I don't think so. For all the changes that we've currently put in within this PR I don't think there's anything that'll be affected by promotion rules since all the changes are supposed to work with both systems. 

It is when we actually introduce the new type system that we'll have to be careful about this.